### PR TITLE
Drop support for --packages-dir flag

### PR DESCRIPTION
**Pull request by natebosch**
Towards #1952

Drops support for the flag and updates tests that were depending on it
or referring to it, but does not remove the supporting code. That will
be done in a followup.

- Parse but ignore the flag and warn when it is used.
- Drop tests specifically for the packages dir behavior.
- Remove unnecessary references to the flag.
- Drop utility test code and constants for the packages dir.
- Wherever possible replace checks for content within the packages dir
  with checks for content in the package spec, and sometimes cache.
- When working with git shas which are unpredictable and have an ugly
  format in the package spec check instead for lines that contain the
  sha, or which have a package spec which either specificially did or
  did not chagne after an upgrade.

### CODE REVIEW

**Comment by kevmoo** [comment]
@natebosch – need to `dartfmt`

File: test/pub_get_and_upgrade_test.dart [hunk 1]
```diff
@@ -34,57 +34,21 @@ main() {
...
-            [d.file('myapp_name.dart', 'main() => "myapp_name";')])
-      ]).validate();
+      await pubCommand(command);
 
       await d.dir("myapp", [
         d.packagesFile({"myapp_name": "."})
       ]).validate();
     });
 
-    test(
```
Comment (jakemac53) [inline test/pub_get_and_upgrade_test.dart]: Should these be checking the package spec? Or does it just not really matter?

File: test/upgrade/git/upgrade_to_nonexistent_pubspec_test.dart [hunk 2]
```diff
@@ -1,44 +0,0 @@
...
-// for details. All rights reserved. Use of this source code is governed by a
-// BSD-style license that can be found in the LICENSE file.
-
-import 'package:test/test.dart';
-
-import '../../descriptor.dart' as d;
-import '../../test_pub.dart';
-
-main() {
-  test("upgrades Git packages to a nonexistent pubspec", () async {
```
Comment (jakemac53) [inline test/upgrade/git/upgrade_to_nonexistent_pubspec_test.dart]: this test doesn't seem --packages-dir specific?

File: test/pub_get_and_upgrade_test.dart (hunk 1 above)
Comment (natebosch) [reply test/pub_get_and_upgrade_test.dart]: The current behavior is to include the package in the package spec regardless of whether is has a lib directory. I'm assuming that behavior is OK...

File: test/upgrade/git/upgrade_to_nonexistent_pubspec_test.dart (hunk 2 above)
Comment (natebosch) [reply test/upgrade/git/upgrade_to_nonexistent_pubspec_test.dart]: hmm, yeah I am not sure why I thought it was... Restored it

File: lib/src/command/downgrade.dart [hunk 3]
```diff
@@ -27,17 +27,17 @@ class DowngradeCommand extends PubCommand {
...
-    argParser.addFlag('packages-dir',
-        negatable: true,
-        help: "Generate a packages/ directory when installing packages.");
+    argParser.addFlag('packages-dir', negatable: true, hide: true);
   }
 
   Future run() async {
+    if (argResults.wasParsed('packages-dir')) {
+      log.warning(
+          log.yellow('The --packages-dir flag is an unsupported no-op'));
```
Comment (munificent) [inline lib/src/command/downgrade.dart]: "no-op" is jargon. How about:

> The --packages-dir flag is no longer used and does nothing.

Here and below.

File: test/downgrade/doesnt_change_git_dependencies_test.dart [hunk 4]
```diff
@@ -18,23 +18,15 @@ main() {
...
 
-    // TODO(rnystrom): Remove "--packages-dir" and validate using the
-    // ".packages" file instead of looking in the "packages" directory.
-    await pubGet(args: ["--packages-dir"]);
+    await pubGet();
 
-    await d.dir(packagesPath, [
-      d.dir('foo', [d.file('foo.dart', 'main() => "foo";')])
-    ]).validate();
+    final originalFooSpec = packageSpecLine('foo');
```
Comment (munificent) [inline test/downgrade/doesnt_change_git_dependencies_test.dart]: Nit: We don't use "final" for local variables in pub even when we could. I don't think it adds much benefit in general, and I think it's good to be consistent.

File: lib/src/command/downgrade.dart (hunk 3 above)
Comment (natebosch) [reply lib/src/command/downgrade.dart]: Good call. Done

File: test/downgrade/doesnt_change_git_dependencies_test.dart (hunk 4 above)
Comment (natebosch) [reply test/downgrade/doesnt_change_git_dependencies_test.dart]: Done.

**Pull request merged**